plugins/dell-k2/fu-dell-k2-ec.c

```diff
@@ -779,11 +780,16 @@ fu_dell_k2_ec_write_firmware_helper(FuDevice *device,
 					    error))
 				return FALSE;
 
+			/* a buggy device may not send the acknowledgment receipt */
+			skip_dev_ack = ((dev_type == DELL_K2_EC_DEV_TYPE_LAN) &&
+					(j == (fu_chunk_array_length(pages) - 1)));
```

Comment:
I think i'd much rather have the clearer:

    if (dev_type == DELL_K2_EC_DEV_TYPE_LAN && j == fu_chunk_array_length(pages) -1)
        skip_dev_ack = TRUE;

Reply:
Fixed. Also split the work for writing pages to `fu_dell_k2_ec_write_firmware_pages()` to make it clear.